torch/_masked/__init__.py

```diff
@@ -1067,19 +1067,19 @@ def sum(
                 dtype = torch.int64
     dim_ = _canonical_dim(dim, input.ndim)
     mask_input = _combine_input_and_mask(sum, input, mask)
-    if input.layout == torch.strided:
+    if mask_input.layout == torch.strided:
```

Comment:
Why change this?

Reply:
This was changed because the code was updated to support MaskedTensor inputs (in particular, if the input is MaskedTensor, then we want a differentiable `_combine_input_and_mask`, but we don't need one for vanilla tensor/masks. And so it's more accurate to check the `mask_input`'s layout, which is post-processing, rather than the input